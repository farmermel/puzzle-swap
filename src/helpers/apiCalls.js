import { locationKey, sendGridKey } from './apiKeys';


export const getGeoLocation = async () => {
  try { 
    const initialFetch = await fetch(`https://www.googleapis.com/geolocation/v1/geolocate?key=${locationKey}`, {
      body: JSON.stringify({considerIp:'true'}), 
      headers: {
        'content-type': 'application/json'
      },
      method: 'POST',
      mode: 'cors'
    })
    if(initialFetch.status <= 200) {
      const latLngResponse = await initialFetch.json()
      return await getCities(latLngResponse)
    } else {
      throw new Error('Bad status code');
    }
  } catch (error) {
    throw `failed to retrieve geolocation: ${error}`
  }
}

const getCities = async locationResponse => {
  const { lat, lng } = locationResponse.location
  const initialFetch = await fetch(`https://maps.googleapis.com/maps/api/geocode/json?latlng=${lat},${lng}&key=${locationKey}`);
  const cityResponse = await initialFetch.json();
  return cleanCitiesData(cityResponse.results);

  // .then(response => `${getAddressCity(address_components, 'short')}, ${getAddressState(address_components, 'short')}`)
}

const cleanCitiesData = address => {
  const findType = type => type.types[0] === "locality";
  const location = address.map(obj => obj);
  const placeName = location.filter(findType);

  return placeName[0].address_components[0].short_name;
}


// const sgMail = require('@sendgrid/mail');
// sgMail.setApiKey(process.env.sendGridKey);
// const msg = {
//   to: 'test@example.com',
//   from: 'test@example.com',
//   subject: 'Sending with SendGrid is Fun',
//   text: 'and easy to do anywhere, even with Node.js',
//   html: '<strong>and easy to do anywhere, even with Node.js</strong>',
// };
// sgMail.send(msg);
