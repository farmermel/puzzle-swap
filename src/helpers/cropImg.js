/* eslint-disable */
export const getCroppedImg = (image, pixelCrop, fileName) => {

  const canvas = document.createElement('canvas');
  canvas.width = pixelCrop.width;
  canvas.height = pixelCrop.height;
  const ctx = canvas.getContext('2d');
  console.log('image', image)
  console.log('pixelCropxy', pixelCrop.x, pixelCrop.y)
  ctx.drawImage(
    image,
    pixelCrop.x,
    pixelCrop.y,
    pixelCrop.width,
    pixelCrop.height,
    0,
    0,
    pixelCrop.width,
    pixelCrop.height
  );

  return new Promise((resolve, reject) => {
    canvas.toBlob(file => {
      file.name = fileName;
      resolve(file);
    }, 'image/jpeg');
  });
}