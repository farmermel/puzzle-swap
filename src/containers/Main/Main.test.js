import React from 'react';
import { shallow } from 'enzyme';
import { Main,
mapDispatchToProps,
mapStateToProps } from './Main';
import * as apiCalls from '../../helpers/apiCalls';

describe('Main', () => {
  let wrapper;
  let mockResponse
  beforeEach(() => {
    wrapper = shallow(<Main addLocation={jest.fn()}
                            location=''
                            hasErrored={jest.fn()} />, {disableLifecycleMethods: true})
    window.fetch = jest.fn().mockImplementation( url => {
      return Promise.resolve({
        status: 200,
        json: () => Promise.resolve(mockResponse)
      })
    })
  })

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('has default state', () => {
    const expected =  {puzzleFilter: 'all'};
    expect(wrapper.instance().state).toEqual(expected);
  })

  describe('componentDidMount', () => {
    it('calls getGeoLocation', () => {
      apiCalls.getGeoLocation = jest.fn().mockImplementation(() => 'Denver')
      expect(apiCalls.getGeoLocation).not.toHaveBeenCalled()
      wrapper.instance().componentDidMount();
      expect(apiCalls.getGeoLocation).toHaveBeenCalled()
    })

    it('calls addLocation', async () => {
      expect(wrapper.instance().props.addLocation).not.toHaveBeenCalled();
      await wrapper.instance().componentDidMount();
      expect(wrapper.instance().props.addLocation).toHaveBeenCalled();
    })

    it('calls hasErrored with error message if getGeoLocation fails', async () => {
      window.fetch = jest.fn().mockImplementation(() => Promise.reject({
        status: 500,
        message: 'you\'re a genius'
      }))
      apiCalls.getGeoLocation = jest.fn().mockImplementation(() => window.fetch());
      await wrapper.instance().componentDidMount();
      expect(wrapper.instance().props.hasErrored).toHaveBeenCalledWith('you\'re a genius');
    })
  })

  // describe('handleChange', () => {
  //   it('sets state with event passed in', () => {
  //     const mockEvent = {
  //       target: {
  //         name: 'Casey',
  //         value: 'very high'
  //       }
  //     }

  //     wrapper.instance().handleChange(mockEvent);
  //     expect(wrapper.instance().state.Casey).toEqual('very high');
  //   })
  // })

  describe('mapStateToProps', () => {
    it('maps state to props', () => {
      const mockStore = {
        location: 'Denver',
        user: 'Casey'
      }
      const mapped = mapStateToProps(mockStore);
      expect(mapped).toEqual(mockStore)
    })
  })

  describe('mapDispatchToProps', () => {
    it('maps dispatch to props', () => {
      const mockDispatch = jest.fn();
      const mapped = mapDispatchToProps(mockDispatch);
      expect(mockDispatch).not.toHaveBeenCalled();
      mapped.addLocation();
      expect(mockDispatch).toHaveBeenCalled();
    })
  })
})