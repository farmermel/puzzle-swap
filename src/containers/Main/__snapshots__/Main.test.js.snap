// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    addLocation={[Function]}
    location=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <p
          className="view-description"
>
          Viewing 
          <select
                    id="puzzle-filter"
                    name="puzzleFilter"
                    onChange={[Function]}
          >
                    <option
                              value="all"
                    >
                              all
                    </option>
                    <option
                              value="300-750"
                    >
                              300-750
                    </option>
                    <option
                              value="1000"
                    >
                              1000
                    </option>
                    <option
                              value="1100 or more"
                    >
                              1100 or more
                    </option>
          </select>
           piece puzzles posted in 
          
</p>,
        <Connect(PuzzleContainer) />,
      ],
      "className": "main",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Viewing ",
            <select
              id="puzzle-filter"
              name="puzzleFilter"
              onChange={[Function]}
>
              <option
                            value="all"
              >
                            all
              </option>
              <option
                            value="300-750"
              >
                            300-750
              </option>
              <option
                            value="1000"
              >
                            1000
              </option>
              <option
                            value="1100 or more"
              >
                            1100 or more
              </option>
</select>,
            " piece puzzles posted in ",
            "",
          ],
          "className": "view-description",
        },
        "ref": null,
        "rendered": Array [
          "Viewing ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  value="all"
>
                  all
</option>,
                <option
                  value="300-750"
>
                  300-750
</option>,
                <option
                  value="1000"
>
                  1000
</option>,
                <option
                  value="1100 or more"
>
                  1100 or more
</option>,
              ],
              "id": "puzzle-filter",
              "name": "puzzleFilter",
              "onChange": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "all",
                  "value": "all",
                },
                "ref": null,
                "rendered": "all",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "300-750",
                  "value": "300-750",
                },
                "ref": null,
                "rendered": "300-750",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1000",
                  "value": "1000",
                },
                "ref": null,
                "rendered": "1000",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1100 or more",
                  "value": "1100 or more",
                },
                "ref": null,
                "rendered": "1100 or more",
                "type": "option",
              },
            ],
            "type": "select",
          },
          " piece puzzles posted in ",
          "",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <p
            className="view-description"
>
            Viewing 
            <select
                        id="puzzle-filter"
                        name="puzzleFilter"
                        onChange={[Function]}
            >
                        <option
                                    value="all"
                        >
                                    all
                        </option>
                        <option
                                    value="300-750"
                        >
                                    300-750
                        </option>
                        <option
                                    value="1000"
                        >
                                    1000
                        </option>
                        <option
                                    value="1100 or more"
                        >
                                    1100 or more
                        </option>
            </select>
             piece puzzles posted in 
            
</p>,
          <Connect(PuzzleContainer) />,
        ],
        "className": "main",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Viewing ",
              <select
                id="puzzle-filter"
                name="puzzleFilter"
                onChange={[Function]}
>
                <option
                                value="all"
                >
                                all
                </option>
                <option
                                value="300-750"
                >
                                300-750
                </option>
                <option
                                value="1000"
                >
                                1000
                </option>
                <option
                                value="1100 or more"
                >
                                1100 or more
                </option>
</select>,
              " piece puzzles posted in ",
              "",
            ],
            "className": "view-description",
          },
          "ref": null,
          "rendered": Array [
            "Viewing ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    value="all"
>
                    all
</option>,
                  <option
                    value="300-750"
>
                    300-750
</option>,
                  <option
                    value="1000"
>
                    1000
</option>,
                  <option
                    value="1100 or more"
>
                    1100 or more
</option>,
                ],
                "id": "puzzle-filter",
                "name": "puzzleFilter",
                "onChange": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "all",
                    "value": "all",
                  },
                  "ref": null,
                  "rendered": "all",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "300-750",
                    "value": "300-750",
                  },
                  "ref": null,
                  "rendered": "300-750",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "1000",
                    "value": "1000",
                  },
                  "ref": null,
                  "rendered": "1000",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "1100 or more",
                    "value": "1100 or more",
                  },
                  "ref": null,
                  "rendered": "1100 or more",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            " piece puzzles posted in ",
            "",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
